app/src/main/java/fi/metropolia/movesense/view/measure/MeasureView.kt

```diff
@@ -84,25 +97,32 @@ fun MeasureView(
                     horizontalAlignment = Alignment.CenterHorizontally,
                     modifier = Modifier.fillMaxSize()
                 ) {
-                    Row(modifier = Modifier
-                        .weight(1f)
-                        .fillMaxWidth()) {
+                    Row(
+                        modifier = Modifier
+                            .weight(1f)
+                            .fillMaxWidth()
+                    ) {
                         ShowAnimation(assetName = "animations/48244-dashboard-data-visualization.json")
                     }
-                        Text(
-                            stringResource(id = R.string.loading),
-                            textAlign = TextAlign.Center,
-                            modifier = Modifier
-                                .weight(1f)
-                                .fillMaxWidth(),
-                        )
-                    }
+                    Text(
+                        stringResource(id = R.string.loading),
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier
+                            .weight(1f)
+                            .fillMaxWidth(),
+                    )
                 }
+            }
         }
     )
     if (address != null) {
         LaunchedEffect(Unit) {
             measureViewModel.connect(address)
         }
+        DisposableEffect(Unit) {
+            onDispose {
+                measureViewModel.disconnect(address)
+            }
+        }
```

Comment:
Mielenkiinnosta kysyisin, että oletko huomannut tämän disposablen toimivan? Jäin asiaa hieman epäilemään johtuen siitä, että navigaatio pitää jonkinnäköistä muistia yllä.

Reply:
Vaikuttaisi toimivan ihan ok pienen testauksen perusteella.